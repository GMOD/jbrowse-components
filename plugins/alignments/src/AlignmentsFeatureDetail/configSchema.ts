import { ConfigurationSchema } from '@jbrowse/core/configuration'

export const configSchema = ConfigurationSchema('AlignmentsFeatureWidget', {})
