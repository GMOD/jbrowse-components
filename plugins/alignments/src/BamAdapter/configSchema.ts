import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'
import { types } from 'mobx-state-tree'

export default types.late(() =>
  ConfigurationSchema(
    'BamAdapter',
    {
      bamLocation: {
        type: 'fileLocation',
        defaultValue: { uri: '/path/to/my.bam' },
      },
      index: ConfigurationSchema('BamIndex', {
        indexType: {
          model: types.enumeration('IndexType', ['BAI', 'CSI']),
          type: 'stringEnum',
          defaultValue: 'BAI',
        },
        location: {
          type: 'fileLocation',
          defaultValue: { uri: '/path/to/my.bam.bai' },
        },
      }),
      chunkSizeLimit: {
        type: 'number',
        defaultValue: 100000000,
      },
      fetchSizeLimit: {
        type: 'number',
        defaultValue: 500000000,
      },
      sequenceAdapter: {
        type: 'frozen',
        defaultValue: null,
      },
    },
    { explicitlyTyped: true },
  ),
)
