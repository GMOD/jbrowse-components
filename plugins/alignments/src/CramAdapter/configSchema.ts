import PluginManager from '@jbrowse/core/PluginManager'
import { ConfigurationSchema } from '@jbrowse/core/configuration'

export default (pluginManager: PluginManager) => {
  const { types } = pluginManager.lib['mobx-state-tree']
  return types.late(() =>
    ConfigurationSchema(
      'CramAdapter',
      {
        cramLocation: {
          type: 'fileLocation',
          defaultValue: { uri: '/path/to/my.cram' },
        },
        craiLocation: {
          type: 'fileLocation',
          defaultValue: { uri: '/path/to/my.cram.crai' },
        },
        sequenceAdapter: pluginManager.pluggableConfigSchemaType('adapter'),
      },
      { explicitlyTyped: true },
    ),
  )
}
