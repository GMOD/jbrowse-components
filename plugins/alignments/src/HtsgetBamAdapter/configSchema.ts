import { ConfigurationSchema } from '@jbrowse/core/configuration'
import { types } from 'mobx-state-tree'

export default types.late(() =>
  ConfigurationSchema(
    'HtsgetBamAdapter',
    {
      htsgetBase: {
        type: 'string',
        defaultValue: '',
      },
      htsgetTrackId: {
        type: 'string',
        defaultValue: '',
      },
      sequenceAdapter: {
        type: 'frozen',
        defaultValue: null,
      },
    },
    { explicitlyTyped: true },
  ),
)
