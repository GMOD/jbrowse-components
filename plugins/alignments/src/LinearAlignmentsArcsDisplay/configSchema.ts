import { ConfigurationSchema } from '@jbrowse/core/configuration'
import { linearBasicDisplayConfigSchemaFactory } from '@jbrowse/plugin-linear-genome-view'
import { types, Instance } from 'mobx-state-tree'
import PluginManager from '@jbrowse/core/PluginManager'

/**
 * #config LinearAlignmentsArcsDisplay
 */
function x() {} // eslint-disable-line @typescript-eslint/no-unused-vars

function configSchemaF(pluginManager: PluginManager) {
  // modify config schema to take in a sub coverage display
  return ConfigurationSchema(
    'LinearAlignmentsArcsDisplay',
    {
      /**
       * #slot
       */
      defaultRendering: {
        type: 'stringEnum',
        model: types.enumeration('Rendering', ['pileup']),
        defaultValue: 'pileup',
      },
      /**
       * #slot
       */
      renderers: ConfigurationSchema('RenderersConfiguration', {
        PileupRenderer:
          pluginManager.getRendererType('PileupRenderer').configSchema,
      }),
      /**
       * #slot
       */
      maxFeatureScreenDensity: {
        type: 'number',
        description: 'maximum features per pixel that is displayed in the view',
        defaultValue: 5,
      },

      /**
       * #slot
       */
      colorScheme: {
        type: 'stringEnum',
        model: types.enumeration('colorScheme', [
          'strand',
          'normal',
          'insertSize',
          'insertSizeAndOrientation',
          'mappingQuality',
          'tag',
        ]),
        description: 'color scheme to use',
        defaultValue: 'normal',
      },
    },
    {
      /**
       * #baseConfiguration
       */
      baseConfiguration: linearBasicDisplayConfigSchemaFactory(pluginManager),
      explicitlyTyped: true,
    },
  )
}

export type LinearAlignmentsArcsDisplayConfigModel = ReturnType<
  typeof configSchemaF
>
export type LinearAlignmentsArcsDisplayConfig =
  Instance<LinearAlignmentsArcsDisplayConfigModel>
export default configSchemaF
