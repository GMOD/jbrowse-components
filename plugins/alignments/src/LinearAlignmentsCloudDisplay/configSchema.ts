import PluginManager from '@jbrowse/core/PluginManager'
import { ConfigurationSchema } from '@jbrowse/core/configuration'
import { linearBasicDisplayConfigSchemaFactory } from '@jbrowse/plugin-linear-genome-view'
import { types } from 'mobx-state-tree'

/**
 * #config LinearAlignmentsCloudDisplay
 */
function x() {} // eslint-disable-line @typescript-eslint/no-unused-vars

function configSchemaF(pluginManager: PluginManager) {
  return ConfigurationSchema(
    'LinearAlignmentsCloudDisplay',
    {
      /**
       * #slot
       */
      maxFeatureScreenDensity: {
        type: 'number',
        description: 'maximum features per pixel that is displayed in the view',
        defaultValue: 5,
      },

      /**
       * #slot
       */
      colorScheme: {
        type: 'stringEnum',
        model: types.enumeration('colorScheme', [
          'strand',
          'normal',
          'insertSize',
          'insertSizeAndOrientation',
          'mappingQuality',
          'tag',
        ]),
        description: 'color scheme to use',
        defaultValue: 'normal',
      },
    },
    {
      /**
       * #baseConfiguration
       */
      baseConfiguration: linearBasicDisplayConfigSchemaFactory(pluginManager),
      explicitlyTyped: true,
    },
  )
}

export default configSchemaF
