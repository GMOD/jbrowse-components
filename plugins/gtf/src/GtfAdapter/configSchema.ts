import { ConfigurationSchema } from '@jbrowse/core/configuration'

export default ConfigurationSchema(
  'GtfAdapter',
  {
    gtfLocation: {
      type: 'fileLocation',
      defaultValue: { uri: '/path/to/my.gtf', locationType: 'UriLocation' },
    },
  },
  { explicitlyTyped: true },
)
