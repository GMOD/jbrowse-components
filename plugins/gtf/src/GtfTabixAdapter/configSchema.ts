import { types } from 'mobx-state-tree'
import { ConfigurationSchema } from '@jbrowse/core/configuration'

export default ConfigurationSchema(
  'GtfTabixAdapter',
  {
    gtfGzLocation: {
      type: 'fileLocation',
      defaultValue: { uri: '/path/to/my.gtf.gz' },
    },
    index: ConfigurationSchema('GtfTabixIndex', {
      indexType: {
        model: types.enumeration('IndexType', ['TBI', 'CSI']),
        type: 'stringEnum',
        defaultValue: 'TBI',
      },
      location: {
        type: 'fileLocation',
        defaultValue: { uri: '/path/to/my.gtf.gz.tbi' },
      },
    }),
    dontRedispatch: {
      type: 'stringArray',
      defaultValue: ['chromosome', 'region'],
    },
  },
  { explicitlyTyped: true },
)
