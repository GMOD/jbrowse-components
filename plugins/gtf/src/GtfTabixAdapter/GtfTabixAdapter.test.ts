import configSchema from './configSchema'
import Adapter from './GtfTabixAdapter'
import path from 'path'

test('adapter can fetch variants form demo.gtf', async () => {
  const adapter = new Adapter(
    configSchema.create({
      gtfGzLocation: {
        localPath: path.join(
          __dirname,
          '..',
          '..',
          '..',
          'test_data',
          'demo.gtf',
        ),
      },
      index: {
        indexType: 'TBI',
        location: {
          localPath: path.join(
            __dirname,
            '..',
            '..',
            '..',
            'test_data',
            'demo.gtf.tbi',
          ),
        },
      },
    }),
  )
})
