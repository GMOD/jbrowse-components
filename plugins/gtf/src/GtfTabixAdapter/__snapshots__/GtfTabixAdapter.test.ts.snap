// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adapter can fetch features from volvox.sorted.gtf test getfeatures on gtf adapter 1`] = `
Array [
  Object {
    "end": 9000,
    "gene_id": "EDEN",
    "gene_name": "EDEN",
    "name": "EDEN.1",
    "phase": 0,
    "refName": "ctgA",
    "source": "example",
    "start": 1049,
    "strand": 1,
    "subfeatures": Array [
      Object {
        "end": 1500,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.1",
        "parentId": "test-offset-108193",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 1049,
        "strand": 1,
        "transcript_id": "EDEN.1",
        "type": "exon",
        "uniqueId": "test-offset-108193-0",
      },
      Object {
        "end": 1200,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.1",
        "parentId": "test-offset-108193",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 1049,
        "strand": 1,
        "transcript_id": "EDEN.1",
        "type": "five_prime_UTR",
        "uniqueId": "test-offset-108193-1",
      },
      Object {
        "end": 1500,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.1",
        "parentId": "test-offset-108193",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 1200,
        "strand": 1,
        "transcript_id": "EDEN.1",
        "type": "CDS",
        "uniqueId": "test-offset-108193-2",
      },
      Object {
        "end": 3902,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.1",
        "parentId": "test-offset-108193",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 2999,
        "strand": 1,
        "transcript_id": "EDEN.1",
        "type": "CDS",
        "uniqueId": "test-offset-108193-3",
      },
      Object {
        "end": 3902,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.1",
        "parentId": "test-offset-108193",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 2999,
        "strand": 1,
        "transcript_id": "EDEN.1",
        "type": "exon",
        "uniqueId": "test-offset-108193-4",
      },
      Object {
        "end": 5500,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.1",
        "parentId": "test-offset-108193",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 4999,
        "strand": 1,
        "transcript_id": "EDEN.1",
        "type": "CDS",
        "uniqueId": "test-offset-108193-5",
      },
      Object {
        "end": 5500,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.1",
        "parentId": "test-offset-108193",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 4999,
        "strand": 1,
        "transcript_id": "EDEN.1",
        "type": "exon",
        "uniqueId": "test-offset-108193-6",
      },
      Object {
        "end": 7608,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.1",
        "parentId": "test-offset-108193",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 6999,
        "strand": 1,
        "transcript_id": "EDEN.1",
        "type": "CDS",
        "uniqueId": "test-offset-108193-7",
      },
      Object {
        "end": 9000,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.1",
        "parentId": "test-offset-108193",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 6999,
        "strand": 1,
        "transcript_id": "EDEN.1",
        "type": "exon",
        "uniqueId": "test-offset-108193-8",
      },
      Object {
        "end": 9000,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.1",
        "parentId": "test-offset-108193",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 7608,
        "strand": 1,
        "transcript_id": "EDEN.1",
        "type": "three_prime_UTR",
        "uniqueId": "test-offset-108193-9",
      },
    ],
    "transcript_id": "EDEN.1",
    "type": "transcript",
    "uniqueId": "test-offset-108193",
  },
  Object {
    "end": 9000,
    "gene_id": "EDEN",
    "gene_name": "EDEN",
    "name": "EDEN.2",
    "phase": 0,
    "refName": "ctgA",
    "source": "example",
    "start": 1049,
    "strand": 1,
    "subfeatures": Array [
      Object {
        "end": 1500,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.2",
        "parentId": "test-offset-108285",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 1049,
        "strand": 1,
        "transcript_id": "EDEN.2",
        "type": "exon",
        "uniqueId": "test-offset-108285-0",
      },
      Object {
        "end": 1200,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.2",
        "parentId": "test-offset-108285",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 1049,
        "strand": 1,
        "transcript_id": "EDEN.2",
        "type": "five_prime_UTR",
        "uniqueId": "test-offset-108285-1",
      },
      Object {
        "end": 1500,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.2",
        "parentId": "test-offset-108285",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 1200,
        "strand": 1,
        "transcript_id": "EDEN.2",
        "type": "CDS",
        "uniqueId": "test-offset-108285-2",
      },
      Object {
        "end": 5500,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.2",
        "parentId": "test-offset-108285",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 4999,
        "strand": 1,
        "transcript_id": "EDEN.2",
        "type": "CDS",
        "uniqueId": "test-offset-108285-3",
      },
      Object {
        "end": 5500,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.2",
        "parentId": "test-offset-108285",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 4999,
        "strand": 1,
        "transcript_id": "EDEN.2",
        "type": "exon",
        "uniqueId": "test-offset-108285-4",
      },
      Object {
        "end": 7608,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.2",
        "parentId": "test-offset-108285",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 6999,
        "strand": 1,
        "transcript_id": "EDEN.2",
        "type": "CDS",
        "uniqueId": "test-offset-108285-5",
      },
      Object {
        "end": 9000,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.2",
        "parentId": "test-offset-108285",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 6999,
        "strand": 1,
        "transcript_id": "EDEN.2",
        "type": "exon",
        "uniqueId": "test-offset-108285-6",
      },
      Object {
        "end": 9000,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.2",
        "parentId": "test-offset-108285",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 7608,
        "strand": 1,
        "transcript_id": "EDEN.2",
        "type": "three_prime_UTR",
        "uniqueId": "test-offset-108285-7",
      },
    ],
    "transcript_id": "EDEN.2",
    "type": "transcript",
    "uniqueId": "test-offset-108285",
  },
  Object {
    "end": 9000,
    "gene_id": "EDEN",
    "gene_name": "EDEN",
    "name": "EDEN.3",
    "phase": 0,
    "refName": "ctgA",
    "source": "example",
    "start": 1299,
    "strand": 1,
    "subfeatures": Array [
      Object {
        "end": 1500,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.3",
        "parentId": "test-offset-108763",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 1299,
        "strand": 1,
        "transcript_id": "EDEN.3",
        "type": "exon",
        "uniqueId": "test-offset-108763-0",
      },
      Object {
        "end": 1500,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.3",
        "parentId": "test-offset-108763",
        "phase": 1,
        "refName": "ctgA",
        "source": "example",
        "start": 1299,
        "strand": 1,
        "transcript_id": "EDEN.3",
        "type": "five_prime_UTR",
        "uniqueId": "test-offset-108763-1",
      },
      Object {
        "end": 3902,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.3",
        "parentId": "test-offset-108763",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 2999,
        "strand": 1,
        "transcript_id": "EDEN.3",
        "type": "exon",
        "uniqueId": "test-offset-108763-2",
      },
      Object {
        "end": 3300,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.3",
        "parentId": "test-offset-108763",
        "phase": 1,
        "refName": "ctgA",
        "source": "example",
        "start": 2999,
        "strand": 1,
        "transcript_id": "EDEN.3",
        "type": "five_prime_UTR",
        "uniqueId": "test-offset-108763-3",
      },
      Object {
        "end": 3902,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.3",
        "parentId": "test-offset-108763",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 3300,
        "strand": 1,
        "transcript_id": "EDEN.3",
        "type": "CDS",
        "uniqueId": "test-offset-108763-4",
      },
      Object {
        "end": 5500,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.3",
        "parentId": "test-offset-108763",
        "phase": 1,
        "refName": "ctgA",
        "source": "example",
        "start": 4999,
        "strand": 1,
        "transcript_id": "EDEN.3",
        "type": "CDS",
        "uniqueId": "test-offset-108763-5",
      },
      Object {
        "end": 5500,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.3",
        "parentId": "test-offset-108763",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 4999,
        "strand": 1,
        "transcript_id": "EDEN.3",
        "type": "exon",
        "uniqueId": "test-offset-108763-6",
      },
      Object {
        "end": 7600,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.3",
        "parentId": "test-offset-108763",
        "phase": 1,
        "refName": "ctgA",
        "source": "example",
        "start": 6999,
        "strand": 1,
        "transcript_id": "EDEN.3",
        "type": "CDS",
        "uniqueId": "test-offset-108763-7",
      },
      Object {
        "end": 9000,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.3",
        "parentId": "test-offset-108763",
        "phase": 0,
        "refName": "ctgA",
        "source": "example",
        "start": 6999,
        "strand": 1,
        "transcript_id": "EDEN.3",
        "type": "exon",
        "uniqueId": "test-offset-108763-8",
      },
      Object {
        "end": 9000,
        "gene_id": "EDEN",
        "gene_name": "EDEN",
        "name": "EDEN.3",
        "parentId": "test-offset-108763",
        "phase": 1,
        "refName": "ctgA",
        "source": "example",
        "start": 7600,
        "strand": 1,
        "transcript_id": "EDEN.3",
        "type": "three_prime_UTR",
        "uniqueId": "test-offset-108763-9",
      },
    ],
    "transcript_id": "EDEN.3",
    "type": "transcript",
    "uniqueId": "test-offset-108763",
  },
  Object {
    "end": 11500,
    "name": "Apple1",
    "phase": 0,
    "refName": "ctgA",
    "source": "bare_predicted",
    "start": 9999,
    "strand": 1,
    "subfeatures": Array [
      Object {
        "end": 11500,
        "name": "Apple1",
        "parentId": "test-offset-110832",
        "phase": 0,
        "refName": "ctgA",
        "source": "bare_predicted",
        "start": 9999,
        "strand": 1,
        "transcript_id": "Apple1",
        "type": "CDS",
        "uniqueId": "test-offset-110832-0",
      },
    ],
    "transcript_id": "Apple1",
    "type": "transcript",
    "uniqueId": "test-offset-110832",
  },
  Object {
    "end": 17000,
    "name": "cds-Apple2",
    "phase": 0,
    "refName": "ctgA",
    "source": "predicted",
    "start": 13499,
    "strand": 1,
    "subfeatures": Array [
      Object {
        "end": 13800,
        "name": "cds-Apple2",
        "parentId": "test-offset-110898",
        "phase": 0,
        "refName": "ctgA",
        "source": "predicted",
        "start": 13499,
        "strand": 1,
        "transcript_id": "cds-Apple2",
        "type": "CDS",
        "uniqueId": "test-offset-110898-0",
      },
      Object {
        "end": 15500,
        "name": "cds-Apple2",
        "parentId": "test-offset-110898",
        "phase": 1,
        "refName": "ctgA",
        "source": "predicted",
        "start": 14999,
        "strand": 1,
        "transcript_id": "cds-Apple2",
        "type": "CDS",
        "uniqueId": "test-offset-110898-1",
      },
      Object {
        "end": 17000,
        "name": "cds-Apple2",
        "parentId": "test-offset-110898",
        "phase": 2,
        "refName": "ctgA",
        "source": "predicted",
        "start": 16499,
        "strand": 1,
        "transcript_id": "cds-Apple2",
        "type": "CDS",
        "uniqueId": "test-offset-110898-2",
      },
    ],
    "transcript_id": "cds-Apple2",
    "type": "transcript",
    "uniqueId": "test-offset-110898",
  },
  Object {
    "end": 23000,
    "name": "rna-Apple3",
    "phase": 0,
    "refName": "ctgA",
    "source": "exonerate",
    "start": 17399,
    "strand": 1,
    "subfeatures": Array [
      Object {
        "end": 17999,
        "name": "rna-Apple3",
        "parentId": "test-offset-111093",
        "phase": 0,
        "refName": "ctgA",
        "source": "exonerate",
        "start": 17399,
        "strand": 1,
        "transcript_id": "rna-Apple3",
        "type": "UTR",
        "uniqueId": "test-offset-111093-0",
      },
      Object {
        "end": 18800,
        "name": "rna-Apple3",
        "parentId": "test-offset-111093",
        "phase": 0,
        "refName": "ctgA",
        "source": "exonerate",
        "start": 17399,
        "strand": 1,
        "transcript_id": "rna-Apple3",
        "type": "exon",
        "uniqueId": "test-offset-111093-1",
      },
      Object {
        "end": 18800,
        "name": "rna-Apple3",
        "parentId": "test-offset-111093",
        "phase": 0,
        "refName": "ctgA",
        "source": "exonerate",
        "start": 17999,
        "strand": 1,
        "transcript_id": "rna-Apple3",
        "type": "CDS",
        "uniqueId": "test-offset-111093-2",
      },
      Object {
        "end": 19500,
        "name": "rna-Apple3",
        "parentId": "test-offset-111093",
        "phase": 0,
        "refName": "ctgA",
        "source": "exonerate",
        "start": 18999,
        "strand": 1,
        "transcript_id": "rna-Apple3",
        "type": "CDS",
        "uniqueId": "test-offset-111093-3",
      },
      Object {
        "end": 19500,
        "name": "rna-Apple3",
        "parentId": "test-offset-111093",
        "phase": 0,
        "refName": "ctgA",
        "source": "exonerate",
        "start": 18999,
        "strand": 1,
        "transcript_id": "rna-Apple3",
        "type": "exon",
        "uniqueId": "test-offset-111093-4",
      },
      Object {
        "end": 21200,
        "name": "rna-Apple3",
        "parentId": "test-offset-111093",
        "phase": 0,
        "refName": "ctgA",
        "source": "exonerate",
        "start": 20999,
        "strand": 1,
        "transcript_id": "rna-Apple3",
        "type": "CDS",
        "uniqueId": "test-offset-111093-5",
      },
      Object {
        "end": 23000,
        "name": "rna-Apple3",
        "parentId": "test-offset-111093",
        "phase": 0,
        "refName": "ctgA",
        "source": "exonerate",
        "start": 20999,
        "strand": 1,
        "transcript_id": "rna-Apple3",
        "type": "exon",
        "uniqueId": "test-offset-111093-6",
      },
      Object {
        "end": 23000,
        "name": "rna-Apple3",
        "parentId": "test-offset-111093",
        "phase": 0,
        "refName": "ctgA",
        "source": "exonerate",
        "start": 21200,
        "strand": 1,
        "transcript_id": "rna-Apple3",
        "type": "UTR",
        "uniqueId": "test-offset-111093-7",
      },
    ],
    "transcript_id": "rna-Apple3",
    "type": "transcript",
    "uniqueId": "test-offset-111093",
  },
]
`;
